### ISSUE TITLE
How to consider eAnnotation name of the attribute of EClass instead of Feature name

### ISSUE THREAD

**Issue by arkumar5** [pyecore/pyecore#29]
We are trying to load xmi file after registering the dynamic ecore meta model as like mentioned in documentation .  

When it loads the file, I am getting this error 
`ValueError: Feature "nW-PACKAGES" is unknown for NETWORK` 

Whereas, in my EClass, there is feature named "nWPACKAGES" linked through extended meta Data Annotation with the name "NW-PACKAGES"

Is there any way to make pyecore to consider annotation name instead of feature name

Below is my EClass 
` <eClassifiers xsi:type="ecore:EClass" name="NETWORK"> `
` <eAnnotations source="http:///org/eclipse/emf/ecore/util/ExtendedMetaData">`
 ` <details key="name" value="NETWORK"/>` 
` <details key="kind" value="elementOnly"/>`
` </eAnnotations>`
` <eStructuralFeatures xsi:type="ecore:EReference" name="nWPACKAGES"  
 Type="#//NWPACKAGESType1"  containment="true" resolveProxies="false">`
` <eAnnotations source="http:///org/eclipse/emf/ecore/util/ExtendedMetaData">`
` <details key="kind" value="element"/>`
` <details key="name" value="NW-PACKAGES"/>`
` <details key="namespace" value="##targetNamespace"/>`
` </eAnnotations>`
` </eStructuralFeatures>`
` </eClassifiers> `

**Comment by aranega** [comment]
Unfortunately, PyEcore does not currently support `ExtendedMetaData`, but it is something that is on the roadmap. 

At the moment, the solution would be to manually rename `nW-PACKAGES` to `nWPACKAGES` in your xmi, but the I'm aware that this is not the solution you want.

I add a subset of `ExtendedMetaData` capabilities support to the blacklog, I will try to put an acceptable version in the next release.

**Comment by arkumar5** [comment]
Will this be solved easier by modifying some py files on generated static meta model from ecore ? If yes, guide me on what needs to be done. 

Thanks Vincent

**Comment by aranega** [comment]
Indeed, if you are using the static version of your metamodel, there is a solution. This solution is almost equivalent to the one for derived attributes. In your generated `.py` where the whole metamodel code is located, you can change this line (I suppose your code look like the following):

```python
# before
class NETWORK(EObject, metaclass=EMetaEClass):
    nWPACKAGES = EReference(containment=True)  # and possibly other options here

# after
class NETWORK(EObject, metaclass=EMetaEClass):
    nWPACKAGES = EReference(name='nW-PACKAGES', containment=True)  # This mod is useful for XMI serialization

setattr(NETWORK, 'nw-PACKAGES', PACKAGES.nWPACKAGES)  # this mod if useful for XMI deserialization
```

This modification can also be made for `EAttribute` and `x..*` relationships.

**Comment by arkumar5** [comment]
The above solution is also similar to manually modifying the feature name.  The problem is , I have huge model with 500+ attributes on different EClasses (different tag name than feature name). 

I think, I can wait for your proper solution to be implemented. 

Thanks a lot, Vincent.. Much appreciated.

**Comment by aranega** [comment]
Indeed, with so much attribute, it will be very time consuming. Another solution could be to extend the pyecoregen code generator by adding an extra "task" that would generate the code for these modification in a new `py` file that could be imported. I can try to test this solution if you want, I don't think it will be too complicated to do.

**Comment by aranega** [comment]
Indeed, with so much attribute, it will be very time consuming. Another solution could be to extend the pyecoregen code generator by adding an extra "task" that would generate the code for these modification in a new `py` file that could be imported. I can try to test this solution if you want, I don't think it will be too complicated to do.

**Comment by aranega** [comment]
@arkumar5 I had some times to check this out, and I wrote a small code generator using pyecoregen/pymultigen which generates equivalent code in a separate file. The code is composed of two files, one which define the generator tasks, load the metamodel and launch the generator, the other file is the generation template. I put both in a gist [here](https://gist.github.com/aranega/881559dc4d58baf29d0869a14c8cdb8f). The generated code look like this:

```python
from .library import *

Employee.age.name = 'my-age'
setattr(Employee, 'my-age', Employee.age)
```

And the file can be included in the generated `__init__.py`. I know that's far from perfect, but as I don't know when I will have the time to work on the partial support for extenededmetadata, this solution allows you to go forward and will be easy to modify/delete if the final solution does not work exactly as this one.
